pipeline/runtime/run.go

```diff
@@ -33,31 +33,78 @@ func executeRunStep(ctx context.Context, engine *engine.Engine, r *api.StartStep
 	step.Entrypoint = r.Run.Entrypoint
 
 	optimizationState := types.DISABLED
-	if (len(r.OutputVars) > 0 || len(r.Outputs) > 0) && (len(step.Entrypoint) == 0 || len(step.Command) == 0) {
-		return nil, nil, nil, nil, string(optimizationState), telemetry, fmt.Errorf("output variable should not be set for unset entrypoint or command")
-	}
 
 	enablePluginOutputSecrets := IsFeatureFlagEnabled(ciEnablePluginOutputSecrets, engine, step)
 
 	var outputFile string
 
 	var outputSecretsFile string
 
+	// Always ensure DRONE_OUTPUT is available as an output variable for all steps
+	// Add to output_vars if not already present
+	droneOutputFound := false
+	for _, outputVar := range r.OutputVars {
+		if outputVar == "DRONE_OUTPUT" {
+			droneOutputFound = true
+			break
+		}
+	}
+	if !droneOutputFound {
+		r.OutputVars = append(r.OutputVars, "DRONE_OUTPUT")
+		logrus.WithField("step_id", r.ID).Infoln("Added DRONE_OUTPUT to output_vars")
+	}
+
+	// Add to outputs array if not already present
+	droneOutputInOutputs := false
+	for _, output := range r.Outputs {
+		if output.Key == "DRONE_OUTPUT" {
+			droneOutputInOutputs = true
+			break
+		}
+	}
+	if !droneOutputInOutputs {
+		r.Outputs = append(r.Outputs, &api.OutputV2{
+			Key:   "DRONE_OUTPUT",
+			Value: "DRONE_OUTPUT",
+			Type:  api.OutputTypeString,
+		})
+		logrus.WithField("step_id", r.ID).Infoln("Added DRONE_OUTPUT to outputs array")
+	}
+
+	// Set output file paths - use host path for file operations, container path for environment
+	hostOutputFile := fmt.Sprintf("%s/%s.out", pipeline.GetSharedVolPath(), step.ID)
+	containerOutputFile := fmt.Sprintf("/tmp/engine/%s.out", step.ID)
+	
 	if enablePluginOutputSecrets {
-		outputFile = fmt.Sprintf("%s/%s-output.env", pipeline.GetSharedVolPath(), step.ID)
-		step.Envs["DRONE_OUTPUT"] = outputFile
+		hostOutputFile = fmt.Sprintf("%s/%s-output.env", pipeline.GetSharedVolPath(), step.ID)
+		containerOutputFile = fmt.Sprintf("/tmp/engine/%s-output.env", step.ID)
+		step.Envs["DRONE_OUTPUT"] = containerOutputFile
 
 		outputSecretsFile = fmt.Sprintf("%s/%s-output-secrets.env", pipeline.GetSharedVolPath(), step.ID)
-		step.Envs["HARNESS_OUTPUT_SECRET_FILE"] = outputSecretsFile
+		step.Envs["HARNESS_OUTPUT_SECRET_FILE"] = fmt.Sprintf("/tmp/engine/%s-output-secrets.env", step.ID)
```

Comment:
again same thing here, this change is not doing anything. Can you please explain what's the issue you are trying to solve?